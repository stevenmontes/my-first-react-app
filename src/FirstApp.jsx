const getMessage = (name) => `Hola ${name}`;

const FirstApp = () => {
    return (
        <>
            <h1>{getMessage('Chris')}</h1>
            <p>Soy un subtitulo</p>
        </>
    )
}

export default FirstApp;
