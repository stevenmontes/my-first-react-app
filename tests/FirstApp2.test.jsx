import { render, screen } from "@testing-library/react";
import FirstApp from "../src/FirstApp";

describe('Pruebas en FirstApp', () => {

    const title = 'Hola Soy Goku';
    const subTitle = "Soy un subtitulo";


    test('should match with the snapshot', () => {
        const { container } = render(<FirstApp title={title}></FirstApp>);
        expect(container).toMatchSnapshot();
    });

    test('should show the message "Hola Soy Goku"', () => {
        render(<FirstApp title={title}></FirstApp>);
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('should show the message in a <h1>', () => {
        render(<FirstApp title={title}></FirstApp>);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

    test('should show the subtitle sent by props', () => {
        render(<FirstApp title={title} subTitle={subTitle} />);
        expect(screen.getAllByText(subTitle).length).toBe(1);

    })
});