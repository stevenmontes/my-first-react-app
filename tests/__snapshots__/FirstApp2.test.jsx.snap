// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp should match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola Soy Goku
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Chris Montes
  </p>
</div>
`;
